x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_event_log_data_grid.tsx

```diff
@@ -0,0 +1,338 @@
+    setVisibleColumns,
+    setSortingColumns,
+    onChangeItemsPerPage,
+    onChangePage,
+  } = props;
+
+  const columnVisibilityProps = useMemo(
+    () => ({
+      visibleColumns,
+      setVisibleColumns,
+    }),
+    [visibleColumns, setVisibleColumns]
+  );
+
+  const sortingProps = useMemo(
+    () => ({
+      onSort: setSortingColumns,
+      columns: sortingColumns,
+    }),
+    [setSortingColumns, sortingColumns]
+  );
+
+  const paginationProps = useMemo(
+    () => ({
+      ...pagination,
+      pageSizeOptions,
+      onChangeItemsPerPage,
+      onChangePage,
+    }),
+    [pagination, pageSizeOptions, onChangeItemsPerPage, onChangePage]
+  );
+
+  // Main cell renderer, renders durations, statuses, etc.
+  const renderCell = ({ rowIndex, columnId }: EuiDataGridCellValueElementProps) => {
+    const { pageIndex, pageSize } = pagination;
+    const pagedRowIndex = rowIndex - pageIndex * pageSize;
+
+    const value = logs[pagedRowIndex]?.[columnId as keyof IExecutionLog] as string;
+    return (
+      <RuleEventLogListCellRenderer
+        columnId={columnId as ColumnId}
+        value={value}
+        dateFormat={dateFormat}
+      />
+    );
+  };
+
+  const renderShowingResultsRange = () => {
+    const { pageIndex, pageSize, totalItemCount } = pagination;
+    const end = Math.min(pageSize * (pageIndex + 1), totalItemCount);
+    if (pagination.totalItemCount === 0) {
+      return (
+        <strong>
+          <FormattedMessage
+            id="xpack.triggersActionsUI.sections.ruleDetails.ruleEventLogDataGrid.showingResultsRangeNoResult"
+            defaultMessage="0"
+          />
+        </strong>
+      );
+    }
+    return (
+      <strong>
+        <FormattedMessage
+          id="xpack.triggersActionsUI.sections.ruleDetails.ruleEventLogDataGrid.showingResultsRange"
+          defaultMessage="{start, number} - {end, number}"
+          values={{
+            start: pageIndex * pageSize + 1,
+            end,
+          }}
+        />
+      </strong>
+    );
+  };
+
+  const renderShowingResults = () => {
+    return (
+      <EuiText data-test-subj="ruleEventLogListShowingResults" size="xs">
+        <FormattedMessage
+          id="xpack.triggersActionsUI.sections.ruleDetails.ruleEventLogDataGrid.showingResults"
+          defaultMessage="Showing {range} of {total, number} {type}"
+          values={{
+            range: renderShowingResultsRange(),
+            total: pagination.totalItemCount,
+            type: (
+              <strong>
+                <FormattedMessage
+                  id="xpack.triggersActionsUI.sections.ruleDetails.ruleEventLogDataGrid.showingResultsType"
+                  defaultMessage="log {total, plural, one {entry} other {entries}}"
+                  values={{ total: pagination.totalItemCount }}
+                />
+              </strong>
+            ),
+          }}
+        />
+      </EuiText>
+    );
+  };
+
+  return (
+    <>
+      {renderShowingResults()}
```

Comment:
nit -> maybe this component can become its own component since we are having a lot of table that will have the same needs